core/trino-main/src/main/java/io/trino/memory/LocalMemoryManager.java

```diff
@@ -61,7 +61,11 @@ private void validateHeapHeadroom(NodeMemoryConfig config, long availableMemory)
 
     public MemoryInfo getInfo()
     {
-        return new MemoryInfo(OPERATING_SYSTEM_MX_BEAN.getAvailableProcessors(), memoryPool.getInfo());
+        double systemCpuLoad = 0.0;
+        if (OPERATING_SYSTEM_MX_BEAN instanceof com.sun.management.OperatingSystemMXBean) {
+            systemCpuLoad = ((com.sun.management.OperatingSystemMXBean) OPERATING_SYSTEM_MX_BEAN).getCpuLoad();
+        }
```

Comment:
looking for suggestions here. This could lead to false metrics if `OPERATING_SYSTEM_MX_BEAN instanceof com.sun.management.OperatingSystemMXBean` is always false. The `systemCpuLoad` will always be `0.0` then, and trino-gateway will think there's no load on this cluster

Reply:
I think it's better to make it a jvm requirement if possible. It's also how it's used here: https://github.com/trinodb/trino/blob/17e7b42e59f97a38669ef8753b86e1eb14deb889/core/trino-main/src/main/java/io/trino/server/StatusResource.java#L62